### ISSUE TITLE
404 logic change

### ISSUE THREAD

**Issue by bryancuster** [azu/can-npm-publish#19]
It seems like https://github.com/azu/can-npm-publish/commit/7967d92 changed the way 404s are treated.

For node 12 and 1.3.4 the exist code is 1 and `npm ERR! code E404` is printed to the terminal
For node 14 and 1.3.4 the exist code is 1 and `npm ERR! code E404` is printed to the terminal
For node 16 and 1.3.4 the exist code is 1 and `npm ERR! code E404` is printed to the terminal

For node 12 and 1.3.3 things are all good
For node 14 and 1.3.3 things are all good
For node 16 and 1.3.3 `can-npm-publish/lib/can-npm-publish.js:68:37` throws an `Unexpected end of JSON input`/`undefined` error

**Comment by bryancuster** [comment]
Here is my output for node 16 and 1.3.4:
```
> nvm use 16 
Now using node v16.4.0 (npm v7.18.1)
> npx can-npm-publish@1.3.4 --verbose
npm ERR! code E404
npm ERR! 404 Not Found - GET https://mycompany.com/api/npm/npm/@scoped%2fpkgname- not_found
npm ERR! 404 
npm ERR! 404  '@scoped/pkgname@latest' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
{
  "error": {
    "code": "E404",
    "summary": "Not Found - GET https://mycompany.com/api/npm/npm/@scoped%2fpkgname- not_found",
    "detail": "\n '@scoped/pkgname@latest' is not in the npm registry.\nYou should bug the author to publish it (or use the name yourself!)\n\nNote that you can also install from a\ntarball, folder, http url, or git url."
  }
}

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2021-06-23T17_27_52_658Z-debug.log

> echo $status
1
```

**Comment by azu** [comment]
https://github.com/azu/can-npm-publish/pull/18/files#r649639587
Maybe, this change relates

**Comment by azu** [comment]
npm 7 has a bug…

npm 7

```
❯ npm -v
7.18.1

~/.ghq/github.com/azu/can-npm-publish master*
❯ npm view not_found_package_123 --json 2>stderr.log 1>stdout.log

~/.ghq/github.com/azu/can-npm-publish master*
❯ cat stdout.log

~/.ghq/github.com/azu/can-npm-publish master*
❯ cat stderr.log
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/not_found_package_123 - Not found
npm ERR! 404
npm ERR! 404  'not_found_package_123@latest' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
{
  "error": {
    "code": "E404",
    "summary": "Not Found - GET https://registry.npmjs.org/not_found_package_123 - Not found",
    "detail": "\n 'not_found_package_123@latest' is not in the npm registry.\nYou should bug the author to publish it (or use the name yourself!)\n\nNote that you can also install from a\ntarball, folder, http url, or git url."
  }
}

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/azu/.npm/_logs/2021-06-23T23_35_58_106Z-debug.log
```

npm 6

```
~/.ghq/github.com/azu/can-npm-publish master*
❯ npm -v
6.14.13

~/.ghq/github.com/azu/can-npm-publish master*
❯ npm view not_found_package_123 --json 2>stderr.log 1>stdout.log

~/.ghq/github.com/azu/can-npm-publish master*
❯ cat stdout.log
{
  "error": {
    "code": "E404",
    "summary": "'not_found_package_123' is not in the npm registry.\nYou should bug the author to publish it\n(or use the name yourself!)\n\nNote that you can also install from a\ntarball, folder, http url, or git url.",
    "detail": "\n 'not_found_package_123@latest' is not in the npm registry.\nYou should bug the author to publish it (or use the name yourself!)\n\nNote that you can also install from a\ntarball, folder, http url, or git url."
  }
}

~/.ghq/github.com/azu/can-npm-publish master*
❯ cat stderr.log
npm ERR! code E404
npm ERR! 404 'not_found_package_123' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it
npm ERR! 404 (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
npm ERR! 404
npm ERR! 404  'not_found_package_123@latest' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/azu/.npm/_logs/2021-06-23T23_36_48_205Z-debug.log
```

**Comment by azu** [comment]
Related issues:

- [[BUG] npm view --json sompackage dist-tags has invalid json · Issue #3075 · npm/cli](https://github.com/npm/cli/issues/3075)
- [[BUG] npm view --json outputs illegal JSON for existing package but non-existent version · Issue #3184 · npm/cli](https://github.com/npm/cli/issues/3184)
- [[BUG] NPM 7.x broke the "--json" CLI parameter · Issue #2740 · npm/cli](https://github.com/npm/cli/issues/2740)

**Comment by azu** [comment]
I've fixed this error in #20 
Please try https://github.com/azu/can-npm-publish/releases/tag/v1.3.5


📝 npm 7(Node 16) has a bug in `--json`. #20 includes a workaround, but it may be not complete.
